src/addon/file_server/mod.rs

```diff
@@ -269,8 +285,25 @@ impl<'a> FileServer {
                         return Ok(DirectoryIndex {
                             entries: directory_entries,
                             breadcrumbs,
-                            sort_by_name: false,
-                            sort_by_size: true,
+                            sort: Sort::Size,
+                        });
+                    }
+                    SortBy::DateCreated => {
+                        directory_entries.sort_by_key(|entry| entry.date_created);
+
+                        return Ok(DirectoryIndex {
+                            entries: directory_entries,
+                            breadcrumbs,
+                            sort: Sort::DateCreated,
+                        });
+                    }
+                    SortBy::DateModified => {
+                        directory_entries.sort_by_key(|entry| entry.date_modified);
+
+                        return Ok(DirectoryIndex {
+                            entries: directory_entries,
+                            breadcrumbs,
+                            sort: Sort::DateModified,
```

Comment:
Can we condense this to avoid repetition?

Reply:
Fixed